service-monitor/src/main/java/com/yahoo/vespa/service/monitor/application/DuperModel.java

```diff
@@ -0,0 +1,140 @@
+    private static final ControllerApplication controllerApplication = new ControllerApplication();
+    private static final ControllerHostApplication controllerHostApplication = new ControllerHostApplication();
+
+    private static final Map<ApplicationId, InfraApplication> supportedInfraApplications = Stream.of(
+            configServerApplication,
+            configServerHostApplication,
+            proxyHostApplication,
+            controllerApplication,
+            controllerHostApplication)
+            .collect(Collectors.toMap(InfraApplication::getApplicationId, Function.identity()));
+
+    private final boolean containsInfra;
+    private final boolean useConfigserverConfig;
+
+    // Each of the above infrastructure applications may be active, in case their ApplicationInfo is present here
+    private final ConcurrentHashMap<ApplicationId, ApplicationInfo> infraInfos =
+            new ConcurrentHashMap<>(2 * supportedInfraApplications.size());
+
+    // ApplicationInfo known at construction time
+    private final List<ApplicationInfo> staticInfos = new ArrayList<>();
+
+    @Inject
+    public DuperModel(ConfigserverConfig configServerConfig, FlagSource flagSource) {
+        this(
+                // Whether to include activate infrastructure applications in the DuperModel.
+                new FeatureFlag("dupermodel-contains-infra", true, flagSource).value(),
+
+                // Whether to base the ApplicationInfo for the config server on ConfigserverConfig or InfrastructureProvisioner:
+                // - ConfigserverConfig: The list of config servers comes from VESPA_CONFIGSERVERS environment variable.
+                // - InfrastructureProvisioner: The list of config servers comes from the node repository.
+                //
+                // The goal is to use InfrastructureProvisioner like other infrastructure applications.
+                new FeatureFlag("dupermodel-use-configserverconfig", true, flagSource).value(),
+                configServerConfig.multitenant(),
+                configServerApplication.makeApplicationInfoFromConfig(configServerConfig));
+    }
+
+    /** For testing */
+    public DuperModel(boolean containsInfra,
+                      boolean useConfigserverConfig,
+                      boolean multitenant,
+                      ApplicationInfo configServerApplicationInfo) {
+        this.containsInfra = containsInfra;
+        this.useConfigserverConfig = useConfigserverConfig;
+
+        // Single-tenant applications have the config server as part of the application model, conflicting and would have conflicting.
+        // TODO: Add health monitoring for config server when part of application model.
+        if (useConfigserverConfig && multitenant) {
+            staticInfos.add(configServerApplicationInfo);
+        }
+    }
+
+    public ConfigServerApplication getConfigServerApplication() {
```

Comment:
Seems strange to have these methods, they are only used in tests and can be easily created by the test anyway?

Reply:
If clients of DuperModel needed to get hold of e.g. the ConfigServerApplication, it seems correct to ask the DuperModel for it. There used to be public constants that exposed these applications, but they have been replaced with getters on DuperModel.

These are used in 2 tests. You mean the tests could 'new' the applications, even though the DuperModel already contains them?